src/gfx.cpp

```diff
@@ -946,22 +887,26 @@ uint GetStringListWidth(const StringID *list, FontSize fontsize)
  * @param start_fontsize Font size to start the text with.
  * @return Upper left corner of the glyph associated with the character.
  */
-Point GetCharPosInString(const char *str, const char *ch, FontSize start_fontsize)
+Point GetCharPosInString(std::string_view str, const char *ch, FontSize start_fontsize)
 {
+	/* Ensure "ch" is inside "str" or at the exact end. */
+	assert(ch >= str.data() && (ch - str.data()) <= static_cast<int>(str.size()));
```

Comment:
it would need to become `assert(IsInsideBS(ch, str.data(), str.data() + str.size() + 1));` .. it gets so hard to read :P I will just make it two asserts .. this is already annoyingly hard to read :P Stupid pointers ... :D